frontend/components/utils/Search.js

```diff
@@ -1,60 +1,89 @@
+import { useState, useContext } from "react";
+
 import { Row, Col } from "react-bootstrap";
 
-const Search = props => (
-  <div className="search">
-    <Row noGutters="true">
-      <Col md="1" xs="2" className="align-self-center">
-        <img src="/static/images/icon-search.svg" className="search-icon" />
-      </Col>
-      <Col md="11" xs="10" className="align-self-center">
-        <div className="d-none d-sm-block">
-          <input
-            type="text"
-            className="search-text"
-            placeholder={props.text}
-            onChange={props.handler}
-          />
-        </div>
-        <div className="d-block d-sm-none">
-          <input
-            type="text"
-            className="search-text"
-            placeholder={`${props.text.substring(0, 17)}..`}
-            title={props.text}
-            onChange={props.handler}
-          />
-        </div>
-      </Col>
-    </Row>
-    <style jsx global>{`
-      .search {
-        border-radius: 25px;
-        border: solid 2px #e6e6e6;
-        background-color: #f8f8f8;
-        padding: 4px 14px;
-      }
-
-      .search-icon {
-        width: 14px;
-      }
-
-      .search-text {
-        font-family: BentonSans;
-        font-size: 14px;
-        color: #999999;
-        outline: none;
-        border: 0;
-        background-color: #f8f8f8;
-        width: 100%;
-        font-weight: 100;
-      }
-
-      .search-text::placeholder {
-        color: #999999;
-        opacity: 1; /* Firefox */
-      }
-    `}</style>
-  </div>
-);
+import { DataContext } from "./DataProvider";
+
+const Search = ({ text, handler }) => {
+  const ctx = useContext(DataContext);
+
+  const onChange = state => {
+    ctx.setPagination(pagination => {
+      return {
+        ...pagination,
+        search: state
+      };
+    });
+
+    if (state === null || state === undefined || state.trim().length === 0) {
+      handler(null);
+    }
+  };
+
+  const onSubmit = e => {
+    e.preventDefault();
+    handler(ctx.pagination.search);
+  };
+
+  return (
+    <div className="search">
+      <Row noGutters="true">
+        <Col md="1" xs="2" className="align-self-center">
+          <img src="/static/images/icon-search.svg" className="search-icon" />
+        </Col>
+        <Col md="11" xs="10" className="align-self-center">
+          <div className="d-none d-sm-block">
+            <form onSubmit={onSubmit}>
```

Comment:
1. It is not obvious that you should hit enter (submit the form) to start the search.
2. It is not clear if anything is happening at all once you submit (there is no indication of the process)

Reply:
@frol 
1. Have modified the search box to look like the one used by Google. Users can click on the search icon at the end or press Enter. Exactly what Google follows.

2. Have added a loading overlay for search. Please check the latest commit.